### PR TITLE
Correct outdated fields in `package.json`

### PR DESCRIPTION
**Pull request by smorimoto**
This patch corrects all outdated fields in `package.json` based on the npm
official specification.

Signed-off-by: Sora Morimoto <sora@morimoto.io>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,20 @@
 {
-  "name": "tc39-web-draft",
-  "private": true,
+  "name": "@tc39/tc39.github.io",
```
Comment (ljharb) [inline package.json:2]: since private is true, i'm not sure why the name matters, but this is fine

File: package.json [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
   "version": "0.0.0",
-  "description": "Draft for TC39 website",
-  "main": "index.js",
+  "description": "Get involved in specifying JavaScript",
+  "homepage": "https://github.com/tc39/tc39.github.io",
+  "bugs": {
+    "url": "https://github.com/tc39/tc39.github.io/issues"
+  },
```
Comment (ljharb) [inline package.json]: all of these fields are only used when a package is published, so there's no need for them to be here

File: package.json (hunk 1 above)
Comment (smorimoto) [reply package.json:2]: It was a kind of normalisation.

File: package.json (hunk 2 above)
Comment (smorimoto) [reply package.json]: Indeed. I thought about getting rid of them, but I just made a pure correction here, somehow. On an extreme level, we can remove all the extra metafield except private.

Comment (jasonwilliams) [reply package.json]: Yeah I’m not sure if all of these are needed.

I would change or remove `maintainers` as I believe ujjwal, myself and potentially others are still around too. Yulia also created this site with myself, Ujjwal and some others initially so having the concept of a single “author” doesn’t make much sense in this context.

node we can keep for local development, license is fine, description and name are ok to keep.

Comment (smorimoto) [reply package.json]: Those opinions certainly sound reasonable. I reflected it in the latest commit.

File: package.json [hunk 3]
```diff
@@ -20,19 +17,6 @@
     "prebuild": "npm run sass",
     "start": "npx @11ty/eleventy --serve --watch"
   },
-  "repository": {
```
Comment (ctcpip) [inline package.json:23]: suggest keeping `repository`

Comment (smorimoto) [reply package.json:23]: Given that this is an application, not a package, this field also seems unnecessary. What do you think?

Comment (ljharb) [reply package.json:23]: the repository field still works in applications with `npm repo`, so it's best to keep it.

Comment (smorimoto) [reply package.json:23]: OK - I just restored it

**Review by ctcpip — APPROVED** [review]
ambivalent about `homepage` being nixed, but LGTM

**Pull request merged**